ExampleApp/ExampleApp/Screens/Characteristics/ViewModel/CharacterisiticsViewModel.swift

```diff
@@ -0,0 +1,88 @@
+import CoreBluetooth
+import Foundation
+import RxBluetoothKit
+import RxSwift
+
+class CharacteristicsViewModel: CharacteristicsViewModelType {
+
+    var characteristicsOutput: Observable<[Characteristic]> {
+        return bluetoothService.discoverCharacteristics(for: service)
+    }
+
+    var dataUpdateOutput: Observable<Void> {
+        return characteristicUpdateTrigger.asObservable()
+    }
+
+    var alertTriggerOutput: Observable<String> {
+        return alertTrigger.asObservable()
+    }
+
+    private let disposeBag: DisposeBag = DisposeBag()
+
+    private let bluetoothService: RxBluetoothKitService
+
+    private let service: Service
+
+    private let characteristicUpdateTrigger: PublishSubject<Void> = PublishSubject()
+
+    private let alertTrigger: PublishSubject<String> = PublishSubject()
+
+    private var characteristic: Characteristic?
+
+    private var notificationsDisposable: Disposable?
+
+    init(with bluetoothService: RxBluetoothKitService, service: Service) {
+        self.bluetoothService = bluetoothService
+        self.service = service
+    }
+
+    func setCurrent(characteristic: Characteristic) {
+        self.characteristic = characteristic
+    }
+
+    func triggerValueRead() {
+        guard let characteristic = characteristic else {
+            return
+        }
+
+        characteristic.readValue().subscribe(onSuccess: { [unowned self] char in
+            self.characteristicUpdateTrigger.onNext(Void())
```

Comment:
no need for `Void` can be just `()`

Reply:
You're absolutely right, @bartoszstelmaszuk, it's just my personal preference to use Void. Just looks cleaner to me.